Add new console command "cache:clear:all" using shopware CacheClearer component

raffaelecarelle opened the pull request:
### 1. Why is this change necessary?
Add cache:clear:all automating clear all caches and pools

### 2. What does this change do, exactly?
Use Shopware cache clear to cleaning the caches and pools

### 5. Checklist

- [ ] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [ ] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfill them.

raffaelecarelle committed:
Add CacheClearAllCommand to clear all caches

Introduced a new Symfony console command `cache:clear:all` that clears all caches and pools. Implemented the command and added an integration test to ensure its functionality.

raffaelecarelle committed:
add changelog

raffaelecarelle commented:
@AydinHassan sorry but I accidentally closed another PR

AydinHassan commented on src/Core/Framework/Adapter/Command/CacheClearAllCommand.php:
```diff
@@ -0,0 +1,48 @@
+use Symfony\Component\Console\Output\OutputInterface;
+
+#[Package('core')]
+#[AsCommand(name: 'cache:clear:all', description: 'Clear all caches/pools')]
```
Can you maybe elaborate a little in the description about why this is different to the other options?

raffaelecarelle replied:
Hi @AydinHassan, I added a command help and more complete description

AydinHassan replied:
thanks!

raffaelecarelle committed:
Enhance cache clear command with environment details

Updated the CacheClearAllCommand to include the environment and debug mode in both the description and the output. Added a detailed command help to guide users on how to use it with different environments and debug settings.

AydinHassan commented on tests/integration/Core/Framework/Adapter/Command/CacheClearAllCommandTest.php:
```diff
@@ -0,0 +1,29 @@
+/**
+ * @internal
+ */
+class CacheClearAllCommandTest extends TestCase
```
Can we make this a unit test? Since it's just calling a service I don't think we need an integration test

raffaelecarelle replied:
Done

AydinHassan commented on src/Core/Framework/Adapter/Command/CacheClearAllCommand.php:
```diff
@@ -0,0 +1,66 @@
+
+            $this->cacheClearer->clear();
+
+            if ($output->isVerbose()) {
```
I think we can drop this finished line since we print directly below anyway

raffaelecarelle replied:
done

raffaelecarelle committed:
Refactor CacheClearAllCommand testing

Replaced integration test with a unit test to improve test isolation and efficiency. Removed verbose output from CacheClearAllCommand for simplification.

raffaelecarelle committed:
fix typo

raffaelecarelle committed:
fix checks

AydinHassan commented:
Thanks for the PR @raffaelecarelle it's merged with a few small changes :)
